effekt/shared/src/main/scala/effekt/core/PolymorphismBoxing.scala

```diff
@@ -248,7 +248,7 @@ object PolymorphismBoxing extends Phase[CoreTransformed, CoreTransformed] {
       if (coerce.isIdentity) {
         Stmt.Val(id, transform(tpe), transform(binding), transform(body))
       } else {
-        val orig = TmpValue()
+        val orig = TmpValue("valStatement")
```

Comment:
Same as above. - All the ones in boxing should probably be called the same, as they are all generated for the value before coercion.